common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -334,6 +350,165 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         }
     }
 
+    /**
+     * Makes a call to the signup/start endpoint, mapping responses returned from the server into a command result.
+     * If the call is successful, additionally calls the signup/challenge endpoint, returning the result.
+     */
+    fun signUpStart(parameters: BaseSignUpStartCommandParameters): SignUpStartCommandResult {
+        LogSession.logMethodCall(TAG, "${TAG}.signUpStart")
+        try {
+            val oAuth2Strategy = createOAuth2Strategy(parameters)
+
+            val usePassword = parameters is SignUpStartUsingPasswordCommandParameters
+
+            val signUpStartApiResult = if (usePassword) {
+                performSignUpStartUsingPasswordRequest(
+                    oAuth2Strategy = oAuth2Strategy,
+                    parameters = (parameters as SignUpStartUsingPasswordCommandParameters)
+                )
+            } else {
+                performSignUpStartRequest(
+                    oAuth2Strategy = oAuth2Strategy,
+                    parameters = (parameters as SignUpStartCommandParameters)
+                )
+            }
```

Comment:
We should log which request we route through, password or passwordless

Reply:
I have added logs